src/components/NewEnvironment/index.js

```diff
@@ -0,0 +1,260 @@
+
+  const toggleShowEnvType = () => {
+    inputBranchName !== '' ? setShowEnvType(true) : setShowEnvType(false);
+  }
+  
+  return (
+      <StyledNotificationWrapper>
+        <div className="margins">
+          <Button action={openModal}>
+          <span style={{ display: 'inline-flex', alignItems: 'center', gap: '10px', padding: '0.25em' }}>
+            <span style={{ fontSize: '28px' }}>+</span>
+            <span style={{ fontSize: '16px', lineHeight: '24px' }}>Create Environment</span>
+          </span>
+          </Button>
+        </div>
+        <Modal isOpen={open} onRequestClose={closeModal} contentLabel={`Confirm`} style={customStyles}>
+          <React.Fragment>
+            <Mutation mutation={DEPLOY_ENVIRONMENT_BRANCH_MUTATION} onError={e => console.error(e)}>
+              {(deployEnvironmentBranch, { loading, error, data }) => {
+                if (error) {
+                  return <div>{error.message}</div>;
+                }
+                if (data) {
+                  refresh().then(setClear).then(closeModal);
+                }
+
+                return (
+                    <>
+                      <StyledNewEnvironment>
+                        <div className="env-modal-header">
+                          <span>Create an Environment</span>
+                        </div>
+                        <div className="modal-step">
+                        <span><b>Step 1: </b>Add the branch you wsh to base this environment on, the branch must already exist.</span>
+                          <div className="environment-modal">
+                            <label>
+                              Branch name: <span style={{ color: '#E30000' }}>*</span>
+                            </label>
+                              <input
+                                id="branchName"
+                                className="inputBranch"
+                                type="text"
+                                placeholder="eg Main or Master"
+                                value={inputBranchName}
+                                onChange={setBranchName}
+                                onBlur={() => toggleShowEnvType()}
+                              />
+                          </div>
+                        </div>
+                        { environmentCount <= 0 &&
+                          <div className="modal-step">
+                            <p><b>Step 2: </b>Add this projects Deploy Key to your Git service.</p>
+                            <div className="showHideContainer">
+                              <div className="copy-field">
+                                <div className="field">
+                                  {loadingDK ? <div className="loader"></div>
+                                      : !showDKField ?
+                                          hashValue(dkValue).substring(0, 25)
+                                          : dkValue.length > 75 ? dkValue.substring(0, 50) + "..." : dkValue
```

Comment:
Given that these strings are generally always longer than 75, can we display the whole thing - just for people who like to select/copy/paste - or are we constrained?